### PR TITLE
in edge platforms, Clock freq is hardcoded to 100 in device info. Fixed it by getting from xclbin

### PR DESCRIPTION
**Pull request by chvamshi-xilinx**
Hi,

This change contains a fix to get the clock frequency from xclbin and store it in deviceInfo. This information is used by profiling team  to draw the plots.

Thanks

### CODE REVIEW

**Comment by chvamshi-xilinx** [comment]
Hi @hackwa , Can u please try with this change whether CR-1046162 solved by this fix or not.

**Comment by chvamshi-xilinx** [comment]
retest this please

File: src/runtime_src/core/common/xclbin_parser.cpp [hunk 1]
```diff
@@ -282,6 +292,37 @@ get_softkernels(const axlf* top)
...
+    auto clock_child = xml_project.get_child_optional("project.platform.device.core.kernelClocks"); 
+
+    if(clock_child) { // check whether kernelClocks field exists or not
+      for (auto& xml_clock : xml_project.get_child("project.platform.device.core.kernelClocks")) {
+	if (xml_clock.first != "clock")
+	  continue;
+	auto port = xml_clock.second.get<std::string>("<xmlattr>.port","");
+	auto freq = convert(xml_clock.second.get<std::string>("<xmlattr>.frequency","100"));
+	if(port == "KERNEL_CLK")
+	  kernel_clk_freq = freq;
```
Comment (mamin506) [inline src/runtime_src/core/common/xclbin_parser.cpp]: Use space for indent.

**Comment by mamin506** [comment]
Ideally, we should parser clock freq in zocl and expose it to user space on sysfs.

**Comment by maxzhen** [comment]
@chvamshi-xilinx , please file another pull request to address the comments. I'll merge it now to unblock sprite.

**Pull request merged**